plugin/src/main/groovy/com/novoda/staticanalysis/DefaultViolationsEvaluator.groovy

```diff
@@ -0,0 +1,45 @@
+package com.novoda.staticanalysis
+
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.GradleException
+import org.gradle.api.logging.Logger
+
+class DefaultViolationsEvaluator implements ViolationsEvaluator {
+
+    private final ReportUrlRenderer reportUrlRenderer
+    private final Logger logger
+    private final PenaltyExtension penalty
+
+    DefaultViolationsEvaluator(ReportUrlRenderer reportUrlRenderer, Logger logger, PenaltyExtension penalty) {
+        this.reportUrlRenderer = reportUrlRenderer
+        this.logger = logger
+        this.penalty = penalty
+    }
+
+    @Override
+    void evaluate(Set<Violations> allViolations) {
+        Map<String, Integer> total = [errors: 0, warnings: 0]
+        String fullMessage = '\n'
+        allViolations.each { Violations violations ->
+            int errors = violations.errors
+            int warnings = violations.warnings
+            if (errors > 0 || warnings > 0) {
+                fullMessage += "> ${getViolationsMessage(violations, reportUrlRenderer)}\n"
```

Comment:
Not really important but I was thinking if you separate this `fullMessage` and `total` construction in 2 different loops, it can be simplified a lot. 

```
allViolations.each { violations ->
  fullMessage += ${getViolationsMessage(violations, reportUrlRenderer)} // method adds \n and initial char if necessary
}
total['errors'] = allViolations.collect { it.errors }.sum()
total['warnings'] = allViolations.collect { it.warnings }.sum()
```

Reply:
I wasn't aware of `.sum()`. #TIL

Reply:
I'm not sure if it will work. :) Please double check. Looks cleaner.